pkg/server/interceptors.go

```diff
@@ -52,3 +58,57 @@ func combineInterceptors(a, b grpc.UnaryServerInterceptor) grpc.UnaryServerInter
 		})
 	}
 }
+
+type nodeRequestLog struct {
+	GRPCService string      `json:"grpc_service"`
+	GRPCMethod  string      `json:"grpc_method"`
+	PeerAddress string      `json:"peer_address"`
+	PeerNodeID  string      `json:"peer_node_id"`
+	Msg         interface{} `json:"msg"`
+}
+
+func prepareRequestLog(ctx context.Context, req, server interface{}, methodName string) ([]byte, error) {
+	reqLog := nodeRequestLog{
+		GRPCService: fmt.Sprintf("%T", server),
+		GRPCMethod:  methodName,
+		PeerAddress: "<no peer???>",
+		Msg:         req,
+	}
+	if peer, ok := peer.FromContext(ctx); ok {
+		reqLog.PeerAddress = peer.Addr.String()
+		if peerIdentity, err := identity.PeerIdentityFromPeer(peer); err == nil {
+			reqLog.PeerNodeID = peerIdentity.ID.String()
+		} else {
+			reqLog.PeerNodeID = fmt.Sprintf("<no peer id: %v>", err)
```

Comment:
curious, what's the benefit of this? It seems odd to me to have the peer id possibly be a string with an error.

Reply:
This wouldn't actually set the peer ID; just what gets logged in place of the peer ID if none can be identified.